build/registry/pkg/oci/oci.go

```diff
@@ -286,21 +292,28 @@ func DoUpdateOCIRegistry(ctx context.Context, registryFile string) error {
 
 	reg, err := registry.LoadRegistryFromFile(registryFile)
 	if err != nil {
-		return fmt.Errorf("an error occurred while loading registry entries from file %q: %v", registryFile, err)
+		return nil, fmt.Errorf("an error occurred while loading registry entries from file %q: %v", registryFile, err)
 	}
 
+	artifacts := []registry.ArtifactPushMetadata{}
+
+	// For each plugin in the registry index, look for new ones to be released, and publish them.
 	for _, plugin := range reg.Plugins {
-		if err := handleArtifact(ctx, cfg, &plugin, s3Client, ociClient); err != nil {
-			return err
+		pa, ra, err := handleArtifact(ctx, cfg, &plugin, s3Client, ociClient)
+		if err != nil {
+			return artifacts, err
```

Comment:
Should we return nil? Do we use partial results in case of error?

Reply:
Thanks, @alacuku, I think it would be useful to provide them, as at the end of the day, the previous successful handle runs, if any,  would be successful. WDYT?